File: Backend/src/main/java/app/alertify/controller/dto/summary/v1/CheckGroup.java
package app.alertify.controller.dto.summary.v1;

import java.util.List;
import java.util.Objects;

public class CheckGroup {
    
	private String name;
    private List<Check> checks;
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public List<Check> getChecks() {
		return checks;
	}
	public void setChecks(List<Check> checks) {
		this.checks = checks;
	}
	@Override
	public int hashCode() {
		return Objects.hash(checks, name);
	}
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		CheckGroup other = (CheckGroup) obj;
		return Objects.equals(checks, other.checks) && Objects.equals(name, other.name);
	}
	@Override
	public String toString() {
		return "CheckGroup [name=" + name + ", checks=" + checks + "]";
	}
    
    
}

File: Frontend/src/app/data/check-group.ts
import { Check } from "./check";

export class CheckGroup {
    private _name: string;
    private _checks: Check[];

    constructor(name: string, checks:Check[]) {
        this._name = name;
        this._checks = checks;
    }

    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }
    
    public get checks(): Check[] {
        return this._checks;
    }
    public set checks(value: Check[]) {
        this._checks = value;
    }
}
